spec/models/provider_interface/provider_applications_page_state_spec.rb

```diff
@@ -37,6 +40,26 @@
       expect(page_state.filters.size).to be(expected_number_of_filters)
       expect(headings).not_to include('Provider')
     end
+
+    it 'can return filter config for a list of provider locations' do
+      page_state = described_class.new(params: ActionController::Parameters.new({ provider: [provider1.id] }),
+
+                                       provider_user: another_provider_user)
+
+      headings = page_state.filters.map { |filter| filter[:heading] }
+
+      expect(headings).to include("Locations for #{provider1.name}")
+
+      # TODO: think if more specific test that tests multiple provider_locations
```

Comment:
I think it's ok not to get too into the nitty-gritty of this object — it's tested pretty thoroughly through the view